From d98e22a8586fb86dde49b87b4cb6d507253b3060 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 29 Mar 2025 00:14:54 +0100
Subject: [PATCH] Avoid network access to get Electron runtime checksums

Avoid hitting the network to verify the runtime downloads by providing
them ahead of time. Should fix a build failure on newer versions of
Electron.

See: https://github.com/flathub/tech.feliciano.pocket-casts/pull/11#issuecomment-2758536925
and: https://github.com/flathub/tech.feliciano.pocket-casts/pull/15#issuecomment-2775799726

Co-authored-by: Ian2020
---
 forge.config.js | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/forge.config.js b/forge.config.js
index 468fc4d3556d..a2b08414d54d 100644
--- a/forge.config.js
+++ b/forge.config.js
@@ -1,5 +1,12 @@
 module.exports = {
-	packagerConfig: {},
+	packagerConfig: {
+		"download": {
+			"checksums": {
+				"electron-v34.5.0-linux-x64.zip": "11ba0d7da512d7a9f473e599de7698b7a1760c6afd9a01434ea29ccbfc7975e2",
+				"electron-v34.5.0-linux-arm64.zip": "7eee40339a9ad637c6360edfd02d5e02522432c7b0f0c70cf00cf4684e1b6691"
+			}
+		}
+	},
 	makers: [
 		{
 			name: '@electron-forge/maker-zip'
-- 
2.49.0

